cavern/src/main/java/org/opencadc/cavern/files/PutAction.java

```diff
@@ -183,15 +185,53 @@ public void doAction() throws Exception {
 
             // doing this last because it requires chown which is most likely to fail during experimentation
             log.debug("restore owner & group");
-            UserPrincipal owner = NodeUtil.getOwner(getUpLookupSvc(), node);
-            GroupPrincipal group = NodeUtil.getDefaultGroup(getUpLookupSvc(), owner);
-            NodeUtil.setPosixOwnerGroup(rootPath, target, owner, group);
-            
+            restoreOwnNGroup(rootPath, node);
         } catch (AccessControlException | AccessDeniedException e) {
             log.debug(e);
             syncOutput.setCode(403);
+        } catch (IOException e) {
+            if (e.getMessage().contains("quota exceeded")) {
+                // TODO: Replace the quota limit below with the number of bytes 
+                //       remaining in the quota when we can determine it.
+                // get quota limit from a parent container
```

Comment:
This a good comment, but could you move it closer to where the error message is set?

Reply:
Done.